Recipes.Test/JsonConfiguration/JsonConfigurationTest.cs

```diff
@@ -151,6 +151,223 @@ public void Serializer_deserializes_InheritedTypes()
+            Assert.That(tiger.Name, Is.EqualTo("Ronny"));
+            Assert.That(tiger.Age, Is.EqualTo(5));
+            Assert.That(tiger.TailLength, Is.EqualTo(2));
+            Assert.That(tiger.NumberOfTeeth, Is.EqualTo(50));
+
+            Assert.That(mouse, Is.Not.Null);
+            Assert.That(mouse.Name, Is.EqualTo("Missy"));
+            Assert.That(mouse.Age, Is.EqualTo(7));
+            Assert.That(mouse.TailLength, Is.EqualTo(8));
+            Assert.That(mouse.FurColor, Is.EqualTo(FurColor.Black));
+        }
+
+        [Test]
+        public void Serializer_deserializes_to_type_having_all_constructor_parameters_in_json_when_another_types_constructor_has_all_the_same_parameters_but_one_additional_one_which_is_not_in_the_json()
+        {
+            var dogJson = "{\"name\":\"Barney\",\"furColor\":\"brindle\",\"age\":10}";
+
+            var dog = JsonConvert.DeserializeObject<Animal>(dogJson, JsonConfiguration.DefaultSerializerSettings) as Dog;
+
+            Assert.That(dog, Is.Not.Null);
+            Assert.That(dog.Name, Is.EqualTo("Barney"));
+            Assert.That(dog.Age, Is.EqualTo(10));
+            Assert.That(dog.FurColor, Is.EqualTo(FurColor.Brindle));
+            Assert.That(dog.DogTag, Is.EqualTo("my name is Barney"));
+        }
+
+        [Test]
+        public void Serializer_throws_JsonSerializationException_when_json_deserializes_to_multiple_child_types()
+        {
+            var json = "{\"base\":\"my base string\"}";
+
+            var ex = Assert.Throws<JsonSerializationException>(() => JsonConvert.DeserializeObject<InheritedTypeBase>(json, JsonConfiguration.DefaultSerializerSettings));
+
+            Assert.That(ex.Message, Does.Contain("InheritedType1"));
+            Assert.That(ex.Message, Does.Contain("InheritedType2"));
+        }
+
+        [Test]
+        public void Serializer_throws_JsonSerializationException_when_json_cannot_be_deserialized_into_any_child_types()
+        {
+            var json = "{\"none\":\"none\"}";
+
+            var ex = Assert.Throws<JsonSerializationException>(() => JsonConvert.DeserializeObject<Animal>(json, JsonConfiguration.DefaultSerializerSettings));
+
+            Assert.That(ex.Message, Does.Contain("Unable to deserialize"));
+        }
+
+        [Test]
+        public void Serializer_deserializes_inherited_types_when_null_property_is_not_included_in_json()
+        {
+            var json = "{\"float\":.2,\"int32\":50}";
+
+            var inheritedType3 = JsonConvert.DeserializeObject<IBaseInterface>(json, JsonConfiguration.DefaultSerializerSettings) as InheritedType3;
+
+            Assert.That(inheritedType3, Is.Not.Null);
+            Assert.That(inheritedType3.Int32, Is.EqualTo(50));
+            Assert.That(inheritedType3.Float, Is.EqualTo(.2f));
+            Assert.That(inheritedType3.String, Is.Null);
+        }
+
+        [Test]
+        public void Serializer_throws_JsonSerializationException_when_property_has_been_removed_from_child_type()
```

Comment:
I'm not sure what this is testing that is any different. Maybe title is just throwing me. Is it? ..._when_json_has_extra_properties_not_available_on_any_child_type